### PR TITLE
Support aarch64 in walredo seccomp code

### PR DESCRIPTION
**Pull request by kelvich**
Aarch64 doesn't implement some old syscalls like open and select. Use openat instead of open to check if seccomp is supported. Leave both select and pselect6 in the allowlist since we don't call select syscall directly and may hope that libc will call pselect6 on aarch64.

To check whether some syscall is supported it is possible to use `scmp_sys_resolver` from seccopm package:

```
> apt install seccopm
> scmp_sys_resolver -a x86_64 select
23
> scmp_sys_resolver -a aarch64 select
-10101
> scmp_sys_resolver -a aarch64 pselect6
72
```

Negative value means that syscall is not supported.

Another cross-check is to look up for the actuall syscall table in `unistd.h`. To resolve all the macroses one can use `gcc -E` as it is done in `dump_sys_aarch64()` function in libseccomp/src/arch-syscall-validate.

### CODE REVIEW

**Comment by kelvich** [comment]
should fix https://github.com/neondatabase/neon/issues/3852 and https://github.com/neondatabase/neon/issues/3847

File: pgxn/neon_walredo/seccomp.c [hunk 1]
```diff
@@ -9,6 +9,13 @@
  * To prevent this, it has been decided to limit possible interactions
  * with the outside world using the Secure Computing BPF mode.
  *
+ * This code is intended to support both x86_64 and aarch64. The latter
+ * doesn't implement some syscalls like open and select. So instead of
+ * open we use openat. We add both select (absend on aarch64) and
+ * pselect6 (present on both architectures) to the allowlist. Since we
+ * don't call select syscall directly we may expect that libc will call
+ * pselect6 on aarch64.
+ *
```
Comment (hlinnaka) [inline pgxn/neon_walredo/seccomp.c:18]: We only use open/openat in the tests for whether seccomp is work properly. It's a pretty minor implementation detail, so I think this comment should be moved to a less prominent place. Maybe into  seccomp_load_rules() itself.

It's worth mentioning that we support x86_64 and aarch64, though, and that the system calls are different on different platforms. The set of system calls that we have whitelisted are the same on both platforms, but that's just accidental, not guaranteed.

**Review by hlinnaka — APPROVED** [review]
There's comment in the TODO section in seccomp.c:
> * - Test on any arch other than amd64 to see if it works there.

Please remove/update that too.

File: pgxn/neon_walredo/seccomp.c (hunk 1 above)
Comment (kelvich) [reply pgxn/neon_walredo/seccomp.c:18]: > The set of system calls that we have whitelisted are the same on both platforms

Hm, we use `select` which is not present on `aarch64` and comment talks about that. I'll try to rephrase.

File: pgxn/neon_walredo/seccomp.c [hunk 2]
```diff
@@ -9,6 +9,13 @@
  * To prevent this, it has been decided to limit possible interactions
  * with the outside world using the Secure Computing BPF mode.
  *
+ * This code is intended to support both x86_64 and aarch64. The latter
+ * doesn't implement some syscalls like open and select. We allow both
+ * select (absent on aarch64) and pselect6 (present on both architectures)
+ * Since we don't call select syscall directly we may expect that libc will
+ * call pselect6 on aarch64. One may check syscalls presense on a certain
+ * architecture using `scmp_sys_resolver` tool from seccomp package.
```
Comment (hlinnaka) [inline pgxn/neon_walredo/seccomp.c:17]:
```suggestion
 * We call select(2) through libc, and the libc wrapper calls select or pselect6
 * depending on the architecture. You can check which syscalls are present on
 * different architectures with the `scmp_sys_resolver` tool from the
 * seccomp package.
```

**Review by hlinnaka — APPROVED** [review]

**Pull request merged**